### ISSUE TITLE
Printing maps on React Viewer

### ISSUE THREAD

**Issue by gmtoglia** [HaitiData/haitidata#121]
When the user prints a map using as basemap the "MapBox Satellite", the result is this:
[map (9).pdf](https://github.com/HaitiData/haitidata/files/1430741/map.9.pdf)

When the user prints a map while she/he is 3D viewing the map, the result is this:
[map (10).pdf](https://github.com/HaitiData/haitidata/files/1430752/map.10.pdf)

I am wondering if it's worth it to keep the REACT viewer. Is the other viewer GeoExt.js fully tested and functional (or easier to bug fix)?

**Comment by ppasq** [comment]
1) Printing MapBox basemaps could be a copyright violation. Please create a MapBox account (HaitiData is currently using a temporary one) and check the license if fits your needs (https://www.mapbox.com/tos/#unlawful-and-unauthorized-uses).

2) This is a viewer bug: https://github.com/GeoNode/geonode-client/issues/171

3) GeoExt.js is not easier to fix than the React. It is an external component and as the React viewer bugs should be reported to the respective developers.

**Comment by gmtoglia** [comment]
Ok Paolo,

Can we block the printing before launching and show an alert like “for copyright reasons it’s not possible to print this map”? It’s a momentary patch to buy some time and figure out the bug fix.

**Comment by gmtoglia** [comment]
Dear Paolo, I just found out that on the 26th of October (5 days before opening the issue) we were able to print these maps:

[map (3).pdf](https://github.com/HaitiData/haitidata/files/1471447/map.3.pdf)
[map (6).pdf](https://github.com/HaitiData/haitidata/files/1471444/map.6.pdf)
[map (5).pdf](https://github.com/HaitiData/haitidata/files/1471445/map.5.pdf)
[map (4).pdf](https://github.com/HaitiData/haitidata/files/1471446/map.4.pdf)

As you can see the background issue is not there. Are we sure that is a copyright problem?

**Comment by ppasq** [comment]
No, I meant that printing could be a copyright infringement. It depends on the use of the printed material. Please see more details here: https://www.mapbox.com/tos/#[SttTSttT]

Also, I'd kindly remind you to create your own account if you want to use Mapbox tiles.
